src/modules/hg_branch.rs

```diff
@@ -34,17 +46,27 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
         config.truncation_length as usize
     };
 
-    let branch_name =
-        get_hg_current_bookmark(context).unwrap_or_else(|| get_hg_branch_name(context));
+    let repo = get_hg_repo(context).ok()?;
 
-    let truncated_graphemes = get_graphemes(&branch_name, len);
+    let branch_name = &repo.bookmark.unwrap_or(repo.branch);
+    let truncated_graphemes = get_graphemes(branch_name, len);
+    let truncation_symbol = get_graphemes(config.truncation_symbol, 1);
     // The truncation symbol should only be added if we truncated
-    let truncated_and_symbol = if len < graphemes_len(&branch_name) {
-        let truncation_symbol = get_graphemes(config.truncation_symbol, 1);
+    let truncated_and_symbol = if len < graphemes_len(branch_name) {
         truncated_graphemes + truncation_symbol.as_str()
     } else {
         truncated_graphemes
     };
+    let topic_graphemes = if let Some(topic) = &repo.topic {
+        let truncated_topic_graphemes = get_graphemes(topic, len);
+        if len < graphemes_len(topic) {
+            truncated_topic_graphemes + truncation_symbol.as_str()
+        } else {
+            truncated_topic_graphemes
+        }
+    } else {
+        String::from("")
+    };
```

Comment:
Since you are changing truncation quite a bit, please also try moving it to the `utils::truncate` functions, like the meson module.